pallets/nominee-election/src/lib.rs

```diff
@@ -0,0 +1,294 @@
+        /// Invalid validators feeder
+        BadValidatorsFeeder,
+        /// Validator not found, thus not removed
+        ValidatorNotFound,
+    }
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig {
+        pub coefficients: NomineeCoefficients,
+    }
+
+    #[cfg(feature = "std")]
+    impl Default for GenesisConfig {
+        fn default() -> Self {
+            GenesisConfig {
+                coefficients: NomineeCoefficients {
+                    crf: 100,
+                    nf: 1000,
+                    epf: 10,
+                },
+            }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig {
+        fn build(&self) {
+            Coefficients::<T>::put(self.coefficients.clone());
+        }
+    }
+
+    #[cfg(feature = "std")]
+    impl GenesisConfig {
+        /// Direct implementation of `GenesisBuild::build_storage`.
+        ///
+        /// Kept in order not to break dependency.
+        pub fn build_storage<T: Config>(&self) -> Result<sp_runtime::Storage, String> {
+            <Self as frame_support::traits::GenesisBuild<T>>::build_storage(self)
+        }
+
+        /// Direct implementation of `GenesisBuild::assimilate_storage`.
+        ///
+        /// Kept in order not to break dependency.
+        pub fn assimilate_storage<T: Config>(
+            &self,
+            storage: &mut sp_runtime::Storage,
+        ) -> Result<(), String> {
+            <Self as frame_support::traits::GenesisBuild<T>>::assimilate_storage(self, storage)
+        }
+    }
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {}
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Set nominee score coefficients
+        #[pallet::weight(1000)]
+        #[transactional]
+        pub fn set_coefficients(
+            origin: OriginFor<T>,
+            coefficients: NomineeCoefficients,
+        ) -> DispatchResult {
+            T::UpdateOrigin::ensure_origin(origin)?;
+            let old_coefficients = Self::coefficients();
+            Coefficients::<T>::put(coefficients.clone());
+
+            Self::deposit_event(Event::<T>::CoefficientsUpdated(
+                old_coefficients,
+                coefficients,
+            ));
+            Ok(())
+        }
+
+        /// Set selected validators
+        ///
+        /// If the validators passed are empty, return an error
+        #[pallet::weight(1000)]
+        #[transactional]
+        pub fn set_validators(
+            origin: OriginFor<T>,
+            mut validators: Vec<ValidatorInfo<T::AccountId>>,
+        ) -> DispatchResult {
+            let feeder = ensure_signed(origin)?;
+            ensure!(
+                T::Members::contains(&feeder),
+                Error::<T>::BadValidatorsFeeder
+            );
+            ensure!(!validators.is_empty(), Error::<T>::NoEmptyValidators);
```

Comment:
Could you talk more about this?

Reply:
@yz89 there is a membership of feeders. and if the validators submitted are empty then it'll not clean current list

Reply:
I think there's an extra `!` there

Reply:
sorry, I understand now.